kernel/USER_GUIDE.md

```diff
@@ -0,0 +1,604 @@
+
+
+* Credentials and permissions: Connector must populate its FileSystemClient with necessary configurations and credentials for the client to retrieve the necessary data from the storage system. For example, an implementation based on Hadoop's FileSystem abstractions can be passed S3 credentials via the Hadoop configurations.
+* Decryption: If file system objects are encrypted, then the implementation must decrypt the data before returning the data.
+
+
+#### Implement [ParquetReader](https://delta-io.github.io/delta/snapshot/kernel-api/java/api//io/delta/kernel/client/ParquetHandler.html)
+
+As the name suggests, this interface contains everything related to reading Parquet files. It has been designed such that a connector can plug in a wide variety of implementations, from a simple single-threaded reader, to a very advanced multi-threaded reader with pre-fetching and advanced connector-specific expression pushdown. Let's explore the methods to implement, and the guarantees associated with them. 
+
+
+##### Method contextualizeFileReads() 
+
+This method associates the information of a to-be-read file to a connector-specific context object, called `FileReadContext`. Implementations can use this method to inject connector-specific information to optimize the file read. For example, if you want split reading a 256 MB Parquet file (say, `file1.parquet`) into 2 chunks of size 128 MB, then you can create two successive entries `fileReadContext1(file1.parquet)` and `fileReadContext1(file1.parquet)` where the context objects created by this method contains the byte ranges to read. This byte ranges will be used by the next method to do the necessary partial reads.
+
+**Requirements and guarantees: **Any implementation must adhere to the following guarantees.
+
+
+* The output iterator must maintain the same ordering as the input iterator. For example, if file1 is before file2 in the input iterator, then all the contexts associated with file1 must be before those of file2 in the output iterator.
+
+
+##### Method readParquetFiles() 
+
+This method takes as input information as `FileReadContext`s and returns the data in a series of columnar batches. The columns to be read from the Parquet file are defined by the physical schema, and the return batches must match that schema. To implement this method, you may have to first implement your own `<code>[ColumnarBatch](https://delta-io.github.io/delta/snapshot/kernel-api/java/api//io/delta/kernel/data/ColumnarBatch.html)`</code> and `<code>[ColumnVector](https://delta-io.github.io/delta/snapshot/kernel-api/java/api//io/delta/kernel/data/ColumnVector.html)`</code> which is used to represent the in-memory data generated from the Parquet files.
```

Comment:
This doesn't render correctly:
![image](https://github.com/delta-io/delta/assets/79601771/e4197081-22d1-4ad1-8928-1a5dce1e1464)

I'm not sure what the `<code>` tags are trying to do? Maybe this would work better?
```suggestion
This method takes as input information as `FileReadContext`s and returns the data in a series of columnar batches. The columns to be read from the Parquet file are defined by the physical schema, and the return batches must match that schema. To implement this method, you may have to first implement your own [`ColumnarBatch`](https://delta-io.github.io/delta/snapshot/kernel-api/java/api//io/delta/kernel/data/ColumnarBatch.html) and [`ColumnVector`](https://delta-io.github.io/delta/snapshot/kernel-api/java/api//io/delta/kernel/data/ColumnVector.html) which is used to represent the in-memory data generated from the Parquet files.
```
(tho I'm not sure we even need the link to render as code? seems like just a link is enough)